src/BenchmarkDotNet/Disassemblers/Arm64Disassembler.cs

```diff
@@ -141,45 +141,63 @@ public void Feed(Arm64Instruction instruction)
 
     internal class Arm64Disassembler : ClrMdV2Disassembler
     {
-        // See dotnet/runtime src/coreclr/vm/arm64/thunktemplates.asm/.S for the stub code
-        // ldr  x9, DATA_SLOT(CallCountingStub, RemainingCallCountCell)
-        // ldrh w10, [x9]
-        // subs w10, w10, #0x1
-        private static byte[] callCountingStubTemplate = new byte[12] { 0x09, 0x00, 0x00, 0x58, 0x2a, 0x01, 0x40, 0x79, 0x4a, 0x05, 0x00, 0x71 };
-        // ldr x10, DATA_SLOT(StubPrecode, Target)
-        // ldr x12, DATA_SLOT(StubPrecode, MethodDesc)
-        // br x10
-        private static byte[] stubPrecodeTemplate = new byte[12] { 0x4a, 0x00, 0x00, 0x58, 0xec, 0x00, 0x00, 0x58, 0x40, 0x01, 0x1f, 0xd6 };
-        // ldr x11, DATA_SLOT(FixupPrecode, Target)
-        // br  x11
-        // ldr x12, DATA_SLOT(FixupPrecode, MethodDesc)
-        private static byte[] fixupPrecodeTemplate = new byte[12] { 0x0b, 0x00, 0x00, 0x58, 0x60, 0x01, 0x1f, 0xd6, 0x0c, 0x00, 0x00, 0x58 };
-
-        static Arm64Disassembler()
+        internal sealed class RuntimeSpecificData
         {
-            // The stubs code depends on the current OS memory page size, so we need to update the templates to reflect that
-            int pageSizeShifted = Environment.SystemPageSize / 32;
-            // Calculate the ldr x9, #offset instruction with offset based on the page size
-            callCountingStubTemplate[1] = (byte)(pageSizeShifted & 0xff);
-            callCountingStubTemplate[2] = (byte)(pageSizeShifted >> 8);
+            // See dotnet/runtime src/coreclr/vm/arm64/thunktemplates.asm/.S for the stub code
+            // ldr  x9, DATA_SLOT(CallCountingStub, RemainingCallCountCell)
+            // ldrh w10, [x9]
+            // subs w10, w10, #0x1
+            internal byte[] callCountingStubTemplate = new byte[12] { 0x09, 0x00, 0x00, 0x58, 0x2a, 0x01, 0x40, 0x79, 0x4a, 0x05, 0x00, 0x71 };
+            // ldr x10, DATA_SLOT(StubPrecode, Target)
+            // ldr x12, DATA_SLOT(StubPrecode, MethodDesc)
+            // br x10
+            internal byte[] stubPrecodeTemplate = new byte[12] { 0x4a, 0x00, 0x00, 0x58, 0xec, 0x00, 0x00, 0x58, 0x40, 0x01, 0x1f, 0xd6 };
+            // ldr x11, DATA_SLOT(FixupPrecode, Target)
+            // br  x11
+            // ldr x12, DATA_SLOT(FixupPrecode, MethodDesc)
+            internal byte[] fixupPrecodeTemplate = new byte[12] { 0x0b, 0x00, 0x00, 0x58, 0x60, 0x01, 0x1f, 0xd6, 0x0c, 0x00, 0x00, 0x58 };
+            internal int stubPageSize;
```

Comment:
nit: the fields are set in ctor, so they could be `readonly`

```suggestion
            internal readonly byte[] callCountingStubTemplate = new byte[12] { 0x09, 0x00, 0x00, 0x58, 0x2a, 0x01, 0x40, 0x79, 0x4a, 0x05, 0x00, 0x71 };
            // ldr x10, DATA_SLOT(StubPrecode, Target)
            // ldr x12, DATA_SLOT(StubPrecode, MethodDesc)
            // br x10
            internal readonly byte[] stubPrecodeTemplate = new byte[12] { 0x4a, 0x00, 0x00, 0x58, 0xec, 0x00, 0x00, 0x58, 0x40, 0x01, 0x1f, 0xd6 };
            // ldr x11, DATA_SLOT(FixupPrecode, Target)
            // br  x11
            // ldr x12, DATA_SLOT(FixupPrecode, MethodDesc)
            internal readonly byte[] fixupPrecodeTemplate = new byte[12] { 0x0b, 0x00, 0x00, 0x58, 0x60, 0x01, 0x1f, 0xd6, 0x0c, 0x00, 0x00, 0x58 };
            internal readonly int stubPageSize;
```